File: Rules.cs
namespace GameOfLife;

public class Rules
{
    public Status NextState(Status currentStatus, int livingNeighbors)
    {
        if (livingNeighbors == 3)
        {
            return Status.Alive;
        }
        if (currentStatus == Status.Alive && livingNeighbors == 2)
        {
            return Status.Alive;
        }
        return Status.Dead;
    }
}

File: gol/rules.go
package gol

type CellState int64

const (
	Alive CellState = iota
	Dead
)

func NextState(c CellState, n int64) CellState {
	if n == 3 {
		return Alive
	}
	if n == 2 {
		return c
	}
	return Dead
}
